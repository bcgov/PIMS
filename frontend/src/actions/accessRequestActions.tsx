import * as ActionTypes from 'constants/actionTypes';
import { IPagedItems, IAccessRequest } from 'interfaces';

// User API actions

export interface IStoreAccessRequestsAction {
  type: typeof ActionTypes.STORE_ACCESS_REQUESTS;
  pagedAccessRequests: IPagedItems;
}

/**
 * Store the paged access requests to redux.
 * @param pagedAccessRequests - An page of access requests
 */
export const storeAccessRequests = (pagedAccessRequests: IStoreAccessRequestsAction) => ({
  type: ActionTypes.STORE_ACCESS_REQUESTS,
  pagedAccessRequests: pagedAccessRequests,
});

export interface IStoreAccessRequestAction {
  type: typeof ActionTypes.STORE_ACCESS_REQUEST;
  accessRequest: IAccessRequest;
}

/**
 * Store the single access request to redux.
 * @param accessRequest - a single access request
 */
export const storeAccessRequest = (accessRequest: IStoreAccessRequestAction) => ({
  type: ActionTypes.STORE_ACCESS_REQUEST,
  accessRequest: accessRequest,
});
