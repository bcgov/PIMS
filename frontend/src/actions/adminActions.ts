import * as ActionTypes from 'constants/actionTypes';
import {
  IPagedItems,
  IUser,
  IUserDetails,
  IAgency,
  IRole,
  IUsersFilter,
  IAgencyRecord,
  IAgencyDetail,
} from 'interfaces';
import { IUserRecord } from 'features/admin/users/interfaces/IUserRecord';
import { TableSort } from 'components/Table/TableSort';

//Admin API actions
export interface IStoreUsersAction {
  type: typeof ActionTypes.STORE_USERS;
  pagedUsers: IPagedItems<IUser>;
}

export interface IUsersPaginationAction {
  type: typeof ActionTypes.STORE_USERS;
  pagedUsers: IPagedItems<IUser>;
}

export interface IGetUserAction {
  type: typeof ActionTypes.GET_USER;
  pagedUsers: IUser;
}

export interface IUpdateUserAction {
  type: typeof ActionTypes.UPDATE_USER;
  user: IUser;
}

export interface ISortUsersAction {
  type: typeof ActionTypes.SORT_USERS;
  sort: TableSort<IUserRecord>;
}

export interface IFilterUsersAction {
  type: typeof ActionTypes.FILTER_USERS;
  filter: IUsersFilter;
}

export interface IUsersPageSizeAction {
  type: typeof ActionTypes.SET_USERS_PAGE_SIZE;
  size: number;
}

export interface IUpdateUsersPageIndexAction {
  type: typeof ActionTypes.SET_USERS_PAGE_INDEX;
  pageIndex: number;
}

export interface IStoreUserDetail {
  type: typeof ActionTypes.STORE_USER_DETAILS;
  userDetail: IUserDetails;
}

export const storeUserDetail = (userDetail: IUserDetails) => ({
  type: ActionTypes.STORE_USER_DETAILS,
  // Payload below
  userDetail: userDetail,
});

export const storeUsers = (apiData: IPagedItems<IUser>) => ({
  type: ActionTypes.STORE_USERS,
  pagedUsers: { ...apiData, pageIndex: apiData.page - 1 },
});

export const updateUser = (user: IUser): IUpdateUserAction => ({
  type: ActionTypes.UPDATE_USER,
  user,
});

export const getUsersFilterAction = (filter: IUsersFilter): IFilterUsersAction => ({
  type: ActionTypes.FILTER_USERS,
  filter,
});

export const setUsersPageSize = (size: number) => ({
  type: ActionTypes.SET_USERS_PAGE_SIZE,
  size,
});

export const getUsersPageIndexAction = (pageIndex: number): IUpdateUsersPageIndexAction => ({
  type: ActionTypes.SET_USERS_PAGE_INDEX,
  pageIndex,
});

export const getUsersSortAction = (sort: TableSort<IUserRecord>): ISortUsersAction => ({
  type: ActionTypes.SORT_USERS,
  sort,
});

// agency admin
export interface IStoreAgencyDetail {
  type: typeof ActionTypes.STORE_AGENCY_DETAILS;
  agencyDetail: IAgency;
}

export interface IStoreAgencyAction {
  type: typeof ActionTypes.STORE_AGENCY_RESULTS;
  pagedAgencies: IPagedItems<IAgency>;
}

export interface IAgencyPaginationAction {
  type: typeof ActionTypes.STORE_AGENCY_RESULTS;
  pagedAgencies: IPagedItems<IAgency>;
}

export interface IGetAgencyAction {
  type: typeof ActionTypes.GET_AGENCY;
  pagedAgency: IAgency;
}

export interface IUpdateAgencyAction {
  type: typeof ActionTypes.UPDATE_AGENCY;
  agency: IAgency;
}

export interface ISortAgenciesAction {
  type: typeof ActionTypes.SORT_AGENCIES;
  sort: TableSort<IAgencyDetail>;
}

export const storeAgencyDetails = (agencyDetail: IAgency) => ({
  type: ActionTypes.STORE_AGENCY_DETAILS,
  agencyDetail: agencyDetail,
});

export const storeAgencies = (apiData: IPagedItems<IAgency>) => ({
  type: ActionTypes.STORE_AGENCY_RESULTS,
  pagedAgencies: { ...apiData, pageIndex: apiData.page - 1 },
});

export const updateAgency = (agency: IAgencyDetail): IUpdateAgencyAction => ({
  type: ActionTypes.UPDATE_AGENCY,
  agency,
});

export const setAgenciesPageSize = (size: number) => ({
  type: ActionTypes.SET_AGENCIES_PAGE_SIZE,
  size,
});

export const getAgenciesSortAction = (sort: TableSort<IAgencyRecord>): ISortAgenciesAction => ({
  type: ActionTypes.SORT_AGENCIES,
  sort,
});

export interface IAddNewRoleAndAgency {
  agency: IAgency;
  role: IRole;
}
