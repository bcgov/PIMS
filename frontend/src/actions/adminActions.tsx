import * as ActionTypes from 'constants/actionTypes';

//Admin API actions

export interface IStoreAccessRequestsAction {
  type: typeof ActionTypes.STORE_ACCESS_REQUESTS;
  pagedAccessRequests: IPagedItems;
}

export interface IStoreUsersAction {
  type: typeof ActionTypes.STORE_USERS;
  pagedUsers: IPagedItems;
}

export interface IGetUserAction {
  type: typeof ActionTypes.GET_USER;
  pagedUsers: IUser;
}

export const storeAccessRequests = (pagedAccessRequests: IStoreAccessRequestsAction) => ({
  type: ActionTypes.STORE_ACCESS_REQUESTS,
  pagedAccessRequests: pagedAccessRequests,
});

export interface IStoreUserDetail {
  type: typeof ActionTypes.STORE_USER_DETAILS;
  userDetail: IUserDetails;
}

export const storeUserDetail = (userDetail: IUserDetails) => ({
  type: ActionTypes.STORE_USER_DETAILS,
  // Payload below
  userDetail: userDetail,
});

export const storeUsers = (pagedUsers: IStoreUsersAction) => ({
  type: ActionTypes.STORE_USERS,
  pagedUsers: pagedUsers,
});

export interface IAddNewRoleAndAgency {
  agency: IAgency;
  role: IRole;
}

export interface IAgency {
  id?: number;
}
export interface IRole {
  id?: number;
}
export interface IAccessRequest {
  agencies: IAgency[];
  roles: IRole[];
  id: string;
  user: IUser;
  isGranted?: boolean;
}

export interface IUser {
  id: string;
  displayName?: string;
}

export interface IUserDetails {
  id?: string;
  username: string;
  firstName?: string;
  displayName?: string;
  lastName?: string;
  email?: string;
  isDisabled?: boolean;
  agencies: any[];
  roles: any[];
  createdOn?: string;
  rowVersion?: string;
  note?: string;
}

export interface IPagedItems {
  page: number;
  quantity: number;
  total: number;
  items: [];
}
