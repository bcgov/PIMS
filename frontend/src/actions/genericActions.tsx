import * as ActionTypes from 'constants/actionTypes';

export interface IGenericNetworkAction {
  isFetching: boolean;
  name: string;
  type: string;
  errorMessage?: string;
  status?: number;
  data?: any;
}

export const success = (reducer: string, status?: number, data?: any): IGenericNetworkAction => ({
  isFetching: false,
  name: reducer,
  type: ActionTypes.SUCCESS,
  status,
  data,
});

export const request = (reducer: string): IGenericNetworkAction => ({
  isFetching: true,
  name: reducer,
  type: ActionTypes.REQUEST,
});

export const error = (reducer: string, status?: number, err?: any): IGenericNetworkAction => ({
  isFetching: false,
  name: reducer,
  type: ActionTypes.ERROR,
  errorMessage: err,
  status,
});

export const clear = (reducer: string): IGenericNetworkAction => ({
  isFetching: false,
  name: reducer,
  type: ActionTypes.CLEAR,
});
