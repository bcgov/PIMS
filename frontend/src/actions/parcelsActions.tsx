import * as ActionTypes from 'constants/actionTypes';

//Parcel List API action

export interface IProperty {
  id: number;
  propertyTypeId: 0 | 1; // 0 = Parcel, 1 = Building
  latitude: number;
  longitude: number;
}

export interface IStoreParcelsAction {
  type: typeof ActionTypes.STORE_PARCEL_RESULTS;
  parcelList: IProperty[];
}

export const storeParcelsAction = (parcelList: IProperty[]): IStoreParcelsAction => ({
  type: ActionTypes.STORE_PARCEL_RESULTS,
  parcelList: parcelList,
});

//Parcel Detail API action
export interface IAddress {
  line1: string;
  line2?: string;
  city?: string;
  cityId: number | any;
  province?: string;
  provinceId: string;
  postal: string;
}

export interface IBuilding {
  id: number;
  localId: string;
  description: string;
  address: IAddress;
  latitude: number | any;
  longitude: number | any;
  buildingFloorCount: number | any;
  buildingConstructionType: string | undefined;
  buildingConstructionTypeId: number | any;
  buildingPredominateUse: string | undefined;
  buildingPredominateUseId: string | any;
  buildingOccupantType: string | undefined;
  buildingOccupantTypeId: number | any;
  leaseExpiry?: string;
  occupantName: string;
  transferLeaseOnSale: boolean;
  buildingTenancy: string;
  buildingNetBookValue: number;
  rentableArea: number | any;
  agencyId: number | any;
  evaluations: IEvaluation[];
}

export interface IEvaluation {
  fiscalYear: number | any;
  estimatedValue: number | any;
  assessedValue: number | any;
  netBookValue: number | any;
  appraisedValue: number | any;
}

export interface IParcel {
  id: number | any;
  pid: string;
  pin: string;
  latitude: number | any;
  longitude: number | any;
  statusId: number | any;
  propertyStatus: string | undefined;
  classification: string | undefined;
  classificationId: number | any;
  description: string;
  address: IAddress | any;
  landArea: number | any;
  landLegalDescription: string;
  zoning: string;
  zoningPotential: string;
  agency: string | undefined;
  agencyId: number | any;
  isSensitive: boolean;
  buildings: IBuilding[];
  evaluations: IEvaluation[];
}

export interface IAddress {
  line1: string;
  cityId: number | any;
  provinceId: string;
  postal: string;
}

export interface IParcelDetail {
  propertyTypeId: 0;
  parcelDetail: IParcel | null;
}

export interface IBuildingDetail {
  propertyTypeId: 1;
  parcelDetail: IBuilding | null;
}

export type IPropertyDetail = IParcelDetail | IBuildingDetail;

export interface IStoreParcelDetail {
  type: typeof ActionTypes.STORE_PARCEL_DETAIL;
  parcelDetail: IParcelDetail;
}

export const storeParcelDetail = (parcel: IParcel | null): IStoreParcelDetail => ({
  type: ActionTypes.STORE_PARCEL_DETAIL,
  parcelDetail: {
    propertyTypeId: 0,
    parcelDetail: parcel,
  },
});

export interface IStoreBuildingDetail {
  type: typeof ActionTypes.STORE_BUILDING_DETAIL;
  parcelDetail: IBuildingDetail;
}

export const storeBuildingDetail = (building: IBuilding | null): IStoreBuildingDetail => ({
  type: ActionTypes.STORE_BUILDING_DETAIL,
  parcelDetail: {
    propertyTypeId: 1,
    parcelDetail: building,
  },
});
