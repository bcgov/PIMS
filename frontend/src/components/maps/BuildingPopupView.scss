@import '../../fonts.scss';

.buildingPopup {
  min-width: 211px;
  max-width: 250px;
  box-shadow: 1px 2px 2px;
  text-align: left;
  padding: 10px;

  .row .col :first-child {
    margin-top: 0px;
  }

  .label {
    font-weight: 700;
    padding-right: 5px;
  }
  .name {
    font-weight: bold;
  }

  .list-group {
    margin-top: 6px;
  }
  .list-group-item {
    padding: 0px 0px 5px 0px !important;
    border: none;
    font-family: 'BCSans', Fallback, sans-serif;
    font-size: 12px;
  }
  .menu {
    text-align: center;
  }
  p {
    margin: 0;
  }
  .alert {
    margin: 0;
  }
}
