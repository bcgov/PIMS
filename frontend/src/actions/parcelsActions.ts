import * as ActionTypes from 'constants/actionTypes';
import { PointFeature } from 'components/maps/types';
import { PropertyTypes } from 'constants/propertyTypes';

//Parcel List API action

export interface IProperty {
  id: number | '';
  propertyTypeId?: PropertyTypes;
  agencyId?: number | '';
  latitude: number | '';
  longitude: number | '';
  name?: string;
  description?: string;
  projectNumbers?: string[];
  projectStatus?: string;
  isSensitive: boolean | '';
  createdOn?: string;
  updatedOn?: string;
  updatedByEmail?: string;
  updatedByName?: string;
}

export interface IStoreParcelsAction {
  type: typeof ActionTypes.STORE_PARCEL_RESULTS;
  parcelList: IProperty[];
}

export interface IStoreDraftParcelsAction {
  type: typeof ActionTypes.STORE_DRAFT_PARCEL_RESULTS;
  draftParcelList: PointFeature[];
}

export interface IStoreParcelAction {
  type: typeof ActionTypes.STORE_PARCEL_FROM_MAP_EVENT;
  parcel: IProperty;
}

export const storeDraftParcelsAction = (parcelList: PointFeature[]): IStoreDraftParcelsAction => ({
  type: ActionTypes.STORE_DRAFT_PARCEL_RESULTS,
  draftParcelList: parcelList,
});

export const storeParcelsAction = (parcelList: IProperty[]): IStoreParcelsAction => ({
  type: ActionTypes.STORE_PARCEL_RESULTS,
  parcelList: parcelList,
});

export const storeParcelAction = (parcel: IProperty): IStoreParcelAction => ({
  type: ActionTypes.STORE_PARCEL_FROM_MAP_EVENT,
  parcel,
});

//Parcel Detail API action
export interface IAddress {
  id?: number | undefined;
  line1: string;
  line2?: string;
  administrativeArea: string;
  province?: string;
  provinceId: string;
  postal: string;
}

export enum LeasedLand {
  owned = 0,
  leased = 1,
  other = 2,
}

export interface IBuilding extends IProperty {
  parcelId: number | '';
  pid: number | '';
  address: IAddress;
  buildingFloorCount?: number | '';
  buildingConstructionType?: string;
  buildingConstructionTypeId: number | '';
  buildingPredominateUse?: string;
  buildingPredominateUseId: number | '';
  buildingOccupantType?: string;
  buildingOccupantTypeId: number | '';
  classificationId: number | '';
  classification: string;
  encumbranceReason: string;
  leaseExpiry?: string;
  occupantName: string;
  transferLeaseOnSale: boolean;
  buildingTenancy: string;
  buildingTenancyUpdatedOn?: string;
  rentableArea: number | '';
  totalArea: number | '';
  agencyId: number | '';
  agency: string;
  agencyCode: string;
  subAgency?: string;
  assessedLand: number | '';
  assessedBuilding: number | '';
  evaluations: IEvaluation[];
  fiscals: IFiscal[];
  parcels: IParcel[];
}

export interface IFlatBuilding extends IProperty {
  parcelId: number;
  address: string;
  administrativeArea: string;
  postal: string;
  province: string;
  buildingFloorCount?: number | '';
  buildingConstructionType?: string;
  buildingConstructionTypeId: number | '';
  buildingPredominateUse?: string;
  buildingPredominateUseId: number | '';
  buildingOccupantType?: string;
  buildingOccupantTypeId: number | '';
  classificationId: number | '';
  classification: string;
  leaseExpiry?: string;
  occupantName: string;
  transferLeaseOnSale: boolean;
  buildingTenancy: string;
  rentableArea: number | '';
  agencyId: number | '';
  agency: string;
  agencyCode: string;
  assessedLand: number | '';
  assessedBuilding: number | '';
  netBook: number | '';
  leasedLand: {
    type: LeasedLand;
  };
}

export interface IFiscal {
  parcelId?: number;
  buildingId?: number;
  fiscalYear?: number | '';
  key: string;
  value: number | '';
  rowVersion?: string;
}

export interface IEvaluation {
  parcelId?: number;
  buildingId?: number;
  date?: Date | string;
  key: string;
  firm?: string;
  value: number | '';
  rowVersion?: string;
}

export interface IParcel extends IProperty {
  pid?: string;
  pin?: number | '';
  classification?: string;
  classificationId: number | '';
  encumbranceReason: string;
  address?: IAddress;
  landArea: number | '';
  landLegalDescription: string;
  zoning: string;
  zoningPotential: string;
  agency?: string;
  subAgency?: string;
  agencyId: number | '';
  buildings: IBuilding[];
  parcels: Partial<IParcel[]>;
  assessedLand: number | '';
  assessedBuilding: number | '';
  evaluations: IEvaluation[];
  fiscals: IFiscal[];
  rowVersion?: string;
}

export interface IFlatParcel extends IProperty {
  pid?: string;
  pin?: number | '';
  classification?: string;
  classificationId: number | '';
  address: string;
  administrativeArea: string;
  postal: string;
  landArea: number | '';
  landLegalDescription: string;
  zoning: string;
  zoningPotential: string;
  agency?: string;
  agencyId: number | '';
  isSensitive: boolean;
  buildings: IBuilding[];
  assessedLand: number | '';
  assessedBuilding: number | '';
  netBook: number | '';
}

export interface IParcelDetail {
  propertyTypeId: PropertyTypes;
  parcelDetail: IParcel | null;
  position?: [number, number]; // (optional) a way to override the positioning of the map popup
}

export interface IBuildingDetail {
  propertyTypeId: 1;
  parcelDetail: IBuilding | null;
  position?: [number, number]; // (optional) a way to override the positioning of the map popup
}

export type IPropertyDetail = IParcelDetail | IBuildingDetail;

export interface IStoreParcelDetail {
  type: typeof ActionTypes.STORE_PARCEL_DETAIL;
  parcelDetail: IParcelDetail;
}

export const storeParcelDetail = (
  parcel: IParcel | null,
  position?: [number, number],
): IStoreParcelDetail => {
  return {
    type: ActionTypes.STORE_PARCEL_DETAIL,
    parcelDetail: {
      propertyTypeId: parcel?.propertyTypeId as PropertyTypes,
      parcelDetail: parcel,
      position,
    },
  };
};

export interface IStoreBuildingDetail {
  type: typeof ActionTypes.STORE_BUILDING_DETAIL;
  parcelDetail: IBuildingDetail;
}

export const storeBuildingDetail = (
  building: IBuilding | null,
  position?: [number, number],
): IStoreBuildingDetail => ({
  type: ActionTypes.STORE_BUILDING_DETAIL,
  parcelDetail: {
    propertyTypeId: 1,
    parcelDetail: building,
    position,
  },
});

export interface IStoreAssociatedBuildingDetail {
  type: typeof ActionTypes.STORE_ASSOCIATED_BUILDING_DETAIL;
  associatedBuildingDetail: IBuildingDetail;
}

export const storeAssociatedBuilding = (
  building: IBuilding | null,
  position?: [number, number],
): IStoreAssociatedBuildingDetail => ({
  type: ActionTypes.STORE_ASSOCIATED_BUILDING_DETAIL,
  associatedBuildingDetail: {
    propertyTypeId: 1,
    parcelDetail: building,
    position,
  },
});
