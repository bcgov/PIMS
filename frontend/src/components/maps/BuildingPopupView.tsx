import './BuildingPopupView.scss';

import * as React from 'react';
import { ListGroup, Container } from 'react-bootstrap';
import { IBuilding } from 'actions/parcelsActions';
import { Alert, Row, Col } from 'react-bootstrap';
import { Label } from 'components/common/Label';
import { EvaluationKeys } from '../../constants/evaluationKeys';

export interface IBuildingDetailProps {
  building: IBuilding | null;
}

export const BuildingPopupView: React.FC<IBuildingDetailProps> = ({ building }) => {
  return (
    <Container className="buildingPopup" fluid={true}>
      {!building ? (
        <Alert variant="danger">Failed to load building details.</Alert>
      ) : (
        <Row>
          <Col>
            <ListGroup>
              <ListGroup.Item>
                <Label>Assessed Value: </Label>$
                {building?.evaluations
                  ?.find(e => e.key === EvaluationKeys.Assessed)
                  ?.value?.toLocaleString('en-US', { minimumFractionDigits: 2 })}
              </ListGroup.Item>
            </ListGroup>
            <ListGroup>
              <ListGroup.Item>
                <div>
                  <Label>Address: </Label>
                  {building?.address?.line1}
                </div>
                <div>
                  {building?.address?.city}, {building?.address?.province}{' '}
                  {building?.address?.postal}
                </div>
              </ListGroup.Item>
            </ListGroup>
            <ListGroup>
              <ListGroup.Item>
                <Label>Floor Count: </Label>
                {building?.buildingFloorCount}
              </ListGroup.Item>
              <ListGroup.Item>
                <Label>Predominate Use: </Label>
                {building?.buildingPredominateUse}
              </ListGroup.Item>
              {building.projectNumber && (
                <ListGroup.Item>
                  <Label>RAEG or SPP: </Label>
                  {building?.projectNumber}
                </ListGroup.Item>
              )}
            </ListGroup>
          </Col>
        </Row>
      )}
    </Container>
  );
};
