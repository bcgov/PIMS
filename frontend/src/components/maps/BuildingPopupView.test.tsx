import React from 'react';
import renderer from 'react-test-renderer';
import { BuildingPopupView } from './BuildingPopupView';
import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';
import { IAddress, IBuilding } from 'actions/parcelsActions';
import { useKeycloak } from '@react-keycloak/web';
import { render, cleanup } from '@testing-library/react';
import Claims from 'constants/claims';

jest.mock('@react-keycloak/web');
(useKeycloak as jest.Mock).mockReturnValue({
  keycloak: {
    userInfo: {
      agencies: [1],
    },
    subject: 'test',
  },
});

const history = createMemoryHistory();

const mockBuilding = (agencyId: number) => {
  var building: IBuilding = {
    id: 1,
    parcelId: 1,
    name: 'test name',
    description: 'Test description',
    address: ([
      {
        line1: '1234 Mock Street',
        cityId: 1,
        provinceId: 'BC',
        postal: 'V0S1N0',
      },
    ] as unknown) as IAddress,
    latitude: 50,
    longitude: 50,
    buildingFloorCount: 3,
    buildingConstructionTypeId: 1,
    buildingPredominateUseId: 0,
    buildingOccupantTypeId: 1,
    classification: 'test',
    classificationId: 2,
    occupantName: 'Mock Occupant',
    transferLeaseOnSale: true,
    buildingTenancy: 'string',
    rentableArea: 1,
    agencyId: agencyId,
    agency: 'agency',
    agencyCode: 'agency code',
    evaluations: [],
    fiscals: [
      {
        fiscalYear: 2020,
        key: 'Key',
        value: 2,
      },
    ],
  };
  return building;
};
describe('building popup view', () => {
  afterEach(() => {
    cleanup();
  });
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Router history={history}>
          <BuildingPopupView building={mockBuilding(1)} />
        </Router>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('displays update option when user belongs to buildings agency', () => {
    const { getByText } = render(
      <Router history={history}>
        <BuildingPopupView building={mockBuilding(1)} />
      </Router>,
    );
    expect(getByText(/Update/i)).toBeInTheDocument();
  });

  it('displays view option', () => {
    const { getByText } = render(
      <Router history={history}>
        <BuildingPopupView building={mockBuilding(2)} />
      </Router>,
    );
    expect(getByText(/View/i)).toBeInTheDocument();
  });

  it('always displays update option for SRES', () => {
    (useKeycloak as jest.Mock).mockReturnValue({
      keycloak: {
        userInfo: {
          agencies: [1],
          roles: [Claims.ADMIN_PROPERTIES],
        },
        subject: 'test',
      },
    });
    const { getByText } = render(
      <Router history={history}>
        <BuildingPopupView building={mockBuilding(2)} />
      </Router>,
    );
    expect(getByText(/Update/i)).toBeInTheDocument();
  });
});
