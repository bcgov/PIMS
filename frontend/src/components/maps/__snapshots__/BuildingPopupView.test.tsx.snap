// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`building popup view renders correctly 1`] = `
<div
  className="buildingPopup container-fluid"
>
  <div
    className="row"
  >
    <div
      className="col"
    >
      <div
        className="list-group"
        onKeyDown={[Function]}
      >
        <div
          className="name list-group-item"
          data-rb-event-key={null}
          disabled={false}
          onClick={[Function]}
        >
          test name
        </div>
        <div
          className="list-group-item"
          data-rb-event-key={null}
          disabled={false}
          onClick={[Function]}
        >
          Test description
        </div>
      </div>
      <div
        className="list-group"
        onKeyDown={[Function]}
      >
        <div
          className="list-group-item"
          data-rb-event-key={null}
          disabled={false}
          onClick={[Function]}
        >
          <p
            className="label undefined"
          >
            Assessed Value:
          </p>
          $
        </div>
      </div>
      <div
        className="list-group"
        onKeyDown={[Function]}
      >
        <div
          className="list-group-item"
          data-rb-event-key={null}
          disabled={false}
          onClick={[Function]}
        >
          <div />
          <div>
            ,
             
             
          </div>
        </div>
      </div>
      <div
        className="list-group"
        onKeyDown={[Function]}
      >
        <div
          className="list-group-item"
          data-rb-event-key={null}
          disabled={false}
          onClick={[Function]}
        >
          <p
            className="label undefined"
          >
            Agency:
          </p>
          agency
        </div>
        <div
          className="list-group-item"
          data-rb-event-key={null}
          disabled={false}
          onClick={[Function]}
        >
          <p
            className="label undefined"
          >
            Floor Count:
          </p>
          3
        </div>
        <div
          className="list-group-item"
          data-rb-event-key={null}
          disabled={false}
          onClick={[Function]}
        >
          <p
            className="label undefined"
          >
            Predominate Use:
          </p>
        </div>
      </div>
    </div>
  </div>
  <div
    className="menu row"
  >
    <div
      className="col"
    >
      <a
        href="/mapview/1?disabled=true&loadDraft=false&sidebar=true"
        onClick={[Function]}
      >
        View
      </a>
      <a
        href="/mapview/1?disabled=false&loadDraft=false&sidebar=true"
        onClick={[Function]}
      >
        Update
      </a>
    </div>
  </div>
</div>
`;
